### ISSUE TITLE
EXRLoader - non-image parts

### ISSUE THREAD

**Issue by photopea** [mrdoob/three.js#20195]
I am not able to parse a following EXR file with EXRLoader. It has the bit 9 of flags set - “long names”, and the bit 10 - "nonimage parts". The compression is ZIP and is already supported by EXRLoader.

Could you implement the missing features?

https://drive.google.com/file/d/1fuQXEjaiLu9tUibvVdQKwlr0W289SzKO/view?usp=sharing

**Comment by sciecode** [comment]
Indeed the file used relies on *deep data* and *tile-based* internal representation, both of which are not currently supported. Those have been on my to-do list, however I haven't had enough time to tackle the problem so far. My plan is to work on the following features (in order):

- Tile-based data
- Deep data
- Multi-part file

It might take a while to actually implement these, so I apologize in advance. But I'll do my best to do it sooner rather than later.

**Comment by Mugen87** [comment]
I've recently wanted to load the following EXR file: [test.exr.zip](https://github.com/mrdoob/three.js/files/7653945/test.exr.zip)

But the loader reports the following error: `Something is wrong with PIZ_COMPRESSION BITMAP_SIZE`.

Does the loader throw this message because one of the above features is missing? The file itself can be loaded into GIMP and macOS Preview.

**Comment by sciecode** [comment]
Hey Michael, 

I was testing the file in question and it does not appear the issue is caused by the lack of features mentioned above. It seems to be related with some modifications made to the loader to handle a very weird case where firefox can only support RGBA half-float textures (#16778)

Now the real reason this happens is because we completely ignore number of channels on the input file and try to force the loader to force the result texture to be `RGBAFormat`. In the case of the file you provided, the input file only contains a single channel, which completely breaks the entire logic. I'm surprised this wasn't noticed before to be honest.

Ideally we would need to check if the weird firefox behavior still exists, and if it doesn't, we simply handle the input data correctly. Othewise, I think we need to properly check the cases where only RGB formats need to be expanded to RGBA, and other cases can be decoded normally.

**Comment by Mugen87** [comment]
In general, it's recommended to avoid RGB textures since they might be emulated in software on certain platforms. So I think it's a good approach to convert RGB to RGBA.

I would treat all other cases without such a conversion. For example if an EXR file only defines a single color channel, the resulting texture could use `THREE.RedFormat` (resulting in e.g. `gl.R8`).

**Comment by sciecode** [comment]
My thoughts exactly, however, there is one problem.

> For example if an EXR file only defines a single color channel, the resulting texture could use `THREE.RedFormat` (resulting in `gl.R8`, `gl.R16F` or `gl.R32F` depending on the data type).

These internalFormats are only available in WebGL2 contexts, for WebGL1 we are limited to using `Luminance` and `Alpha` internal formats, which only support UnsignedByType. So we do have other this other limitations to be mindful of, not entirely sure what the current policy for handling these sort of GL1/GL2 support issues.

**Comment by Mugen87** [comment]
Indeed, that's important to keep in mind! Considering the maturity of WebGL 2 I would say it's okay if certain loaders produce texture formats that can only be used with WebGL2.

@mrdoob What do you think about this?

**Comment by sciecode** [comment]
In midst of working on a solution for the problem, I noticed there are more complications. Seems like simply using WebGL2 internal formats might no be enough. One of the side-effects of using `THREE.RedFormat` is that it actually only applies the input data to the red channel of the texture. so we don't actually get the original luminance behavior from the underlying texture. According to [Khronos Image Format docs](https://www.khronos.org/opengl/wiki/Image_Format):

![image](https://user-images.githubusercontent.com/3927951/144769822-79bacb4c-2142-44ff-9526-b6e96a4cdc74.png)

`GL_INTENSITY` is not available in any WebGL contexts and `GL_LUMINANCE` only accepts `UnsignedByteType`, with the extra incovenience of not being color-renderable ( although, this might not be a direct issue ). 

All in all, the only real "solution" is to also treat single-channel images specially. We need to ignore the `outputType` specified by the user and always convert the image to `LuminanceFormat - UInt8`. Which is kind of a bummer, but at least we also keep GL1 support.

**Comment by Mugen87** [comment]
> Which is kind of a bummer, because it completely defeats the purpose of using an .EXR 16/32b luminance map in the first place.

That's the reason why I think we should use `THREE.RedFormat`. Besides, devs can patch the shader sampling via `onBeforeCompile()` if necessary. And the new node material opens up even more possibilities in that regard.

**Comment by sciecode** [comment]
> Using the `RedFormat` texture is still valuable, because we make use of all the precision, but it pretty much requires a custom sampling shader to be used correctly.

Sounds good to me. Working on a PR while we wait for other inputs.

**Comment by mrdoob** [comment]
I don't have an opinion on this one. Whatever you guys think is best 👍
